app/routines/update_school_salesforce_info.rb

```diff
@@ -80,6 +80,30 @@ def call
       break if sf_schools.length < BATCH_SIZE
     end
 
+    # Loop through stale schools and update users associated with them to prevent sync issues
+    School.where.not(updated_salesforce_id: nil) do |merged_school|
+      # Let's make sure it wasn't just created
+      if School.find_by(salesforce_id: merged_school.updated_salesforce_id)
+        merged_school.update!(updated_salesforce_id: nil)
+        next
+      end
+
+      # Find the new school and update the users attached to it
+      sf_school = OpenStax::Salesforce::Remote::School.find_by(name: merged_school.updated_salesforce_id)
+      if sf_school.nil?
+        Sentry.capture_message("Possible merged school not found. Original: #{merged_school.salesforce_id} New: #{merged_school.updated_salesforce_id}")
+      else
+        updated_school = School.find_or_create_by(salesforce_id: sf_school.id)
+        SF_TO_DB_CACHE_COLUMNS_MAP.each do |sf_column, db_column|
+          updated_school.public_send "#{db_column}=", sf_school.public_send(sf_column)
+        end
+        updated_school.save!
+
+        merged_school.users.update_all(school: updated_school) if merged_school.users.any?
+        merged_school.destroy!
```

Comment:
I'd call reload before destroy!, just in case... otherwise it could try to dissociate the users.